text/0000-localizing-api-messages.md

```diff
@@ -0,0 +1,140 @@
+* Sub Group : 
+  * Marc Johnson (marc.johnson@k-int.com)
+  * Julian Ladisch (julian.ladisch@gbv.de)
+  * Peter Murray (peter@indexdata.com)
+  * Tod Olson (tod@uchicago.edu)
+
+# Localizing API (Backend) Messages
+
+## Summary
+  The purpose of this RFC is to define an approach that allows API developer to provide
+messages in the API response depending on the language preference set in the request. 
+
+
+## Motivation
+
+- Help users understand the information coming from the application backend using a language they understand better
+- To allow FOLIO platform to be used by users across the world 
+- To create a tremendous opportunity for growth that we can never could have achieved within just one country.
+- To Promote consistency and make it easier for tools and translators when handling translation files
+
+### In Scope Requirements/Use cases
+- Return localized messages based on the value passed in the accept-language header
+- Handle static messages with placeholder(s)
+
+### Out of Scope Requirements/Use cases
+- [Plural Syntax](https://wiki.folio.org/display/I18N/How+To+translate+FOLIO#HowTotranslateFOLIO-Pluralsyntax)
+- Naming convention to be used for keys
+- Process for managing (decoupled from lokalise.com) translations
+- Returning formatted (HTML/Markdown) messages
+- Usage of [soft hyphen](https://wiki.folio.org/display/I18N/How+To+translate+FOLIO#HowTotranslateFOLIO-Softhyphentobreakwords) to break messages
+- Support customization per tenant
+- Process  for [back porting](https://wiki.folio.org/display/I18N/Backport) API/Backend messages similar to what we have for the front end 
+- Support for [controlled vocabulary](https://issues.folio.org/browse/UXPROD-3148) (runtime data/data coming from DB tables. E.g. Patron Groups for tenants)
+
+## Detailed Explanation/Design
+* Translation file format will depend on the language that is used to develop the module
+  * Java, Groovy - properties
+  * JavaScript, nodejs - json
+  * Any other language - Please contact Technical Council for further guidance
+* Translation files MUST be placed under translations/\<Backend Module Name\>
+* Translation file name should be ll_CC.ext. Here ‘ll’ is an ISO 639 two-letter language code, and ‘CC’ is an ISO 3166 two-letter country code.
+* Translation message keys MUST be a string (alphanumeric characters ONLY)
+* Developer should follow language/framework specific recommended best practices for loading translation at runtime
+* When handling messages with placeholders, replace the placeholders with the actual value on the server side before
+  sending it back to the client
+* STOP using the lang query string parameter.
+* accept-language header value MUST be in ll_CC format. , where ll is a two-letter language code, 
+  and CC is a two-letter country code.
+* When accept-language header is missing or has an incorrect value, return the message value in en_US. If message value 
+  cannot be found, return the message key
```

Comment:
I think it could make it easier to follow the guidance if it was split into 3 sections (please feel free to come up with better names):
* API protocol
* Translation definitions
* Language / Runtime specific